### PR TITLE
Replace label SQL procedures with python procedures

### PR DESCRIPTION
**Pull request by joshelser**
- [x] two test failures to fix

ref #234

### CODE REVIEW

**Comment by rymurr** [comment]
lgtm so far!

**Comment by joshelser** [comment]
Restored the `PredefinedLabels` class and subjected them to the same validation as we do for normal labels (instead of just validating the Condition like we previously did).

File: app/crud/labels.py [hunk 1]
```diff
@@ -214,3 +224,22 @@ def verify_time_fields(cls, time: datetime.datetime) -> datetime.datetime:
...
+        :param session:
+        :return:
+        """
+        df = session.sql("select * from internal.predefined_labels").to_pandas()
+        # Lowercase all of the column names
+        df.rename(
+            columns={col_name: col_name.lower() for col_name in df.axes[1]},
+            inplace=True,
+        )
+        for row in df.to_dict(orient="records"):
```
Comment (joshelser) [inline app/crud/labels.py:243]: Will want to consolidate with changes from #339

**Review by vqmarkman — APPROVED** [review]
Looks good to me. I was a bit skeptical about this change, but my feeling now is that it will be much easier to manage.

File: app/crud/labels.py [hunk 2]
```diff
@@ -214,3 +224,22 @@ def verify_time_fields(cls, time: datetime.datetime) -> datetime.datetime:
     def enabled_or_dynamic(cls, b: bool) -> bool:
         assert isinstance(b, bool)
         return b
+
+
+class PredefinedLabel(Label):
+    @classmethod
+    def validate_all(cls, session: snowflake.snowpark.Session):
```
Comment (rymurr) [inline app/crud/labels.py:231]: can you create a ticket to move the predefined label def'n and creation out of sql and into this?

File: bootstrap/004_labels.sql [hunk 3]
```diff
@@ -263,114 +166,66 @@ END;
...
+from crud import delete_entity
+def delete_label(session, name):
+    return delete_entity(session, 'LABEL', name)
+$$;
 
+-- TODO Differentiate from ungrouped labels  by having "AND IS_DYNAMIC" in the delete clause
 CREATE OR REPLACE PROCEDURE ADMIN.DELETE_DYNAMIC_LABEL(name text)
     RETURNS text
-    LANGUAGE SQL
+    language python
```
Comment (rymurr) [inline bootstrap/004_labels.sql:185]: why have this? Its the same is it not?

File: bootstrap/004_labels.sql [hunk 4]
```diff
@@ -263,114 +166,66 @@ END;
...
-    CALL INTERNAL.UPDATE_LABEL_VIEW();
-    return 'done';
-END;
+$$
+from crud import delete_entity
+def delete_label(session, name):
+    return delete_entity(session, 'LABEL', name)
+$$;
 
 CREATE OR REPLACE PROCEDURE ADMIN.UPDATE_LABEL(oldname text, name text, grp text, rank number, condition text, is_dynamic boolean)
```
Comment (rymurr) [inline bootstrap/004_labels.sql:198]: is it possible to get rid of this one? Do we need the is_dynamic still?

**Review by rymurr — COMMENTED** [review]
looks great, I think we can make a few more simplifcations otherwise :ship: it

File: bootstrap/004_labels.sql (hunk 3 above)
Comment (joshelser) [reply bootstrap/004_labels.sql:185]: I had consolidated the `DELETE_DYNAMIC_LABEL` and `DELETE_LABEL` last weekend, and then regretted doing it in the first pass. I left them separate for this PR, but completely agree that we don't need two long-term.

File: bootstrap/004_labels.sql (hunk 4 above)
Comment (joshelser) [reply bootstrap/004_labels.sql:198]: Same with this one -- intentionally didn't collapse the overloaded functions. If it's good with you, would do that in a follow-on (likely today)

File: app/crud/labels.py (hunk 2 above)
Comment (joshelser) [reply app/crud/labels.py:231]: Of course

File: bootstrap/004_labels.sql (hunk 4 above)
Comment (rymurr) [reply bootstrap/004_labels.sql:198]: foolow on is perfect

File: bootstrap/004_labels.sql (hunk 3 above)
Comment (rymurr) [reply bootstrap/004_labels.sql:185]: foolow on is perfect

**Pull request merged**